verl/experimental/agent_loop/tool_agent_loop.py

```diff
@@ -118,91 +139,119 @@ async def run(self, sampling_params: dict[str, Any], **kwargs) -> AgentLoopOutpu
 
             # no tool calls
             _, tool_calls = await self.tool_parser.extract_tool_calls(response_ids)
+            if self.interaction_config_file:
+                assistant_message = await self.loop.run_in_executor(
+                    None, lambda id=response_ids: self.tokenizer.decode(id)
+                )
+                messages.append({"role": "assistant", "content": assistant_message})
+                if not tool_calls:
+                    user_turns += 1
+                    (
+                        should_terminate_sequence,
+                        interaction_responses,
+                        reward,
+                        metrics,
+                    ) = await interaction.generate_response(request_id, messages, **interaction_kwargs)
+                    turn_scores.append(reward)
+                    responses = [{"role": "user", "content": interaction_responses}]
```

Comment:
I'm a bit confuse here,  `interaction` take effect when no tool_calls, but if no tool_calls, we break the while loop. Then where does this `responses` used?